### PR TITLE
ESMification

### PR DESCRIPTION
**Pull request by 117649**
@onemen 
Make sure you check the review.

### CODE REVIEW

**Comment by onemen** [comment]
@117649 

Thank you.

- can we use these files form Firefox 115 and newer?
- does `BootstrapLoader.jsm:` here match the version you posted on [#292](https://github.com/onemen/TabMixPlus/discussions/292).
- what about `userChrome.jsm` and `xPref.jsm`?

**Comment by 117649** [comment]
> @117649

>     * can we use these files form Firefox 115 and newer?

I think it should. If I'm not making the mistake that 115 already have `Services` in `gloableThis`
But I won't go back just for confirm that. 

>     * does `BootstrapLoader.jsm:` here match the version you posted on [#292](https://github.com/onemen/TabMixPlus/discussions/292).

Basically same. I went further to make `signedState` also unchangeable by make it a getter without setter.

>     * what about `userChrome.jsm` and `xPref.jsm`?

When I have time. They are not big.

File: installation-folder/config.js [hunk 1]
```diff
@@ -8,13 +8,10 @@ try {
...
-  });
-  objRef.AddonSettings = Object.defineProperties({}, temp);
+  const objRef = ChromeUtils.importESModule('resource://gre/modules/addons/XPIDatabase.sys.mjs');
+  objRef.XPIDatabase.SIGNED_TYPES.delete("extension");
 
-  Cu.import('chrome://userchromejs/content/BootstrapLoader.jsm');
+  Services.scriptloader.loadSubScript('chrome://userchromejs/content/BootstrapLoader.jsm');
 } catch (ex) {};
 
 try {
```
Comment (onemen) [inline installation-folder/config.js:17]: @117649 

what about this import `Cu.import('chrome://userchromejs/content/userChrome.jsm');`?

File: chrome/utils/RDFManifestConverter.jsm [hunk 2]
```diff
@@ -3,10 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 "use strict";
 
-var EXPORTED_SYMBOLS = ["InstallRDF"];
-
-ChromeUtils.defineModuleGetter(this, "RDFDataSource",
-                               "chrome://userchromejs/content/RDFDataSource.jsm");
+import {RDFDataSource} from "chrome://userchromejs/content/RDFDataSource.sys.mjs";
```
Comment (onemen) [inline chrome/utils/RDFManifestConverter.jsm:6]: rename RDFDataSource.jsm to RDFDataSource.sys.mjs

it is better to rename all files to .sys.mjs then you can use import everywhere, also in config.js

**Comment by benzBrake** [comment]
Same target, I tried to convert the scripts to sys.mjs 3 days ago  https://github.com/benzBrake/userChrome.js-Loader/tree/ESMification

File: chrome/utils/BootstrapLoader.jsm [hunk 3]
```diff
@@ -6,13 +6,13 @@
 
 let EXPORTED_SYMBOLS = [];
 
-const { XPCOMUtils } = ChromeUtils.import('resource://gre/modules/XPCOMUtils.jsm');
+const { XPCOMUtils } = ChromeUtils.importESModule('resource://gre/modules/XPCOMUtils.sys.mjs');
```
Comment (onemen) [inline chrome/utils/BootstrapLoader.jsm:9]: @117649 

Remove all  `XPCOMUtils` usages.

```js
 XPCOMUtils.defineLazyGetter(sandbox, 'console', () =>
        new ConsoleAPI({ consoleID: `addon/${addon.id}` }));
```

use `ChromeUtils.defineLazyGetter`

**Comment by onemen** [comment]
Thank you @117649 

closing this after fetching the update from https://github.com/xiaoxiaoflood/firefox-scripts

**Pull request closed without merging**